thrift/server.go

```diff
@@ -202,12 +202,16 @@ func (s *Server) handle(origCtx context.Context, handler handler, method string,
 	}
 
 	writer, err = call.Response().Arg3Writer()
+
 	wp = getProtocolWriter(writer)
-	resp.Write(wp.protocol)
-	thriftProtocolPool.Put(wp)
-	err = writer.Close()
+	defer thriftProtocolPool.Put(wp)
+
+	if err := resp.Write(wp.protocol); err != nil {
```

Comment:
(unrelated to this diff but just thought of this) @twilly just as a safety check, let's verify that the passed in protocol here is a `TBinaryProtocol` so that users don't end up calling `Write` with a custom protocol and expect it to work?

Reply:
Agreed. Created an internal ticket to track this.